pallets/creditcoin/src/tests.rs

```diff
@@ -179,10 +194,28 @@ impl TestInfo {
 		TestInfo::default()
 	}
 
+	pub fn with_currency(currency: Currency) -> TestInfo {
+		let default = TestInfo::default();
+
+		TestInfo {
+			loan_terms: LoanTerms {
+				currency: CurrencyId::new::<Test>(&currency),
+				..default.loan_terms
+			},
+			currency,
+			..default
+		}
+	}
+
+	pub fn register_currency(&self) {
+		Currencies::<Test>::insert(CurrencyId::new::<Test>(&self.currency), &self.currency);
```

Comment:
I am not seeing an explicit test calling `register_currency()`. Maybe change this to call the extrinsic and assert_ok instead of writing to storage directly.